src/main/java/io/github/cottonmc/resources/CottonResources.java

```diff
@@ -17,139 +19,146 @@
+	private static final Map<String, ResourceType> BUILTINS = new HashMap<>();
+	private static OreVoteConfig RESOLVED_CONFIG = new OreVoteConfig();
+	
+	@Override
+	public void onInitialize() {
+		builtinMetal("copper", BlockSuppliers.STONE_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("silver", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("lead", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("zinc", BlockSuppliers.STONE_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("aluminum", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("cobalt", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("tin", BlockSuppliers.STONE_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("titanium", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("tungsten", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+
+		builtinMetal("platinum", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("palladium", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("osmium", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinMetal("iridium", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+
+		builtinMetal("steel", null, MACHINE_AFFIXES);
+		builtinMetal("brass", null, MACHINE_AFFIXES);
+		builtinMetal("electrum", null, MACHINE_AFFIXES);
+
+		builtinItem("coal", "dust");
+		BUILTINS.put("coal_coke", new GenericResourceType("coal_coke").withBlockAffix("block", BlockSuppliers.COAL_BLOCK).withItemAffixes(""));
+		builtinItem("mercury");
+
+		builtinItem("iron", "gear", "plate", "dust");
+		builtinItem("gold", "gear", "plate", "dust");
+
+		//These might get rods or molten capsules. They'd just need to be added to the end.
+		builtinRadioactive("uranium", BlockSuppliers.DIAMOND_TIER_ORE, "gear", "plate", "ingot", "nugget");
+		builtinRadioactive("plutonium", null, "gear", "plate", "ingot", "nugget");
+		builtinRadioactive("thorium", null, "gear", "plate", "ingot", "nugget");
+
+		builtinItem("diamond", "gear", "plate", "dust");
+		builtinItem("emerald", "gear", "plate", "dust");
+
+		builtinGem("ruby", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinGem("topaz", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinGem("amethyst", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinGem("peridot", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+		builtinGem("sapphire", BlockSuppliers.IRON_TIER_ORE, MACHINE_AFFIXES);
+
+		for (ResourceType resource : BUILTINS.values()) {
+			OreGenerationSettings oreGenerationSettings = CONFIG.ores.get(resource.getBaseResource());
+			if (oreGenerationSettings == null) {
+				LOGGER.warn("No ore generation settings found for " + resource.getBaseResource() + " so registering anyway");
+				resource.registerAll();
+				//} else if (oreGenerationSettings.enabled) {
+				//    // || IsResourceRequestedBymodJson?
+				//    resource.registerAll();
+			} else {
+				resource.registerAll();
+				nullifyRecipes(resource);
```

Comment:
Why nullify recipes if the ore gen settings are found?

Reply:
I also don't get why there's a `resource.registerAll()` call in both of these. Why not just call it before the if/else?

Reply:
Predates the PR, but yes, at this point the registration can be moved to before the if. We used to skip registration in that case, but interactions with data packs make that difficult.

Honestly deregistering the recipes is even a bit dicey. I'm not sure we want this "if" here. Instead, we should toggle item hiding in REI.